api/src/main/java/org/apache/iceberg/StructTransform.java

```diff
@@ -51,11 +53,16 @@ class StructTransform implements StructLike, Serializable {
     this.transforms = new SerializableFunction[size];
 
     for (int i = 0; i < size; ++i) {
-      int sourceFieldId = fieldTransforms.get(i).sourceFieldId();
+      int[] sourceFieldIds = fieldTransforms.get(i).sourceFieldIds();
       Transform<?, ?> transform = fieldTransforms.get(i).transform();
-      Accessor<StructLike> accessor = schema.accessorForField(sourceFieldId);
+      Accessor<StructLike> accessor = schema.accessorForFields(sourceFieldIds);
```

Comment:
Early question, can we have an array of accessors here?  So this.accessors becomes an array of array, or array of lists.

My thought is, then we can  re-use schema.accessorForField() which has some advantages like caching, and may be simpler than having to implement schema.accessorForFields() and a 'structProjectionAccessor'?

Reply:
One possible way is to add a .bind(Type... types) to the Transform API, to be implemented by multi-arg transforms?

I feel it is more clearer and less heavy-handed than using StructProjection for each row, but definitely would also like to check with @aokolnychyi and @rdblue.  If its not possible, then something like you are doing makes sense.